clang/lib/Sema/SemaChecking.cpp

```diff
@@ -172,6 +172,27 @@ static bool checkArgCount(Sema &S, CallExpr *Call, unsigned DesiredArgCount) {
          << /*is non object*/ 0 << Call->getArg(1)->getSourceRange();
 }
 
+static bool checkBuiltinVerboseTrap(CallExpr *Call, Sema &S) {
```

Comment:
@ahatanak Now that we've changed the interface I think this Sema check should be slightly different.

* We should be checking that the number of arguments is **exactly** 2. This looks like it allows any number.
* The `Category` argument should not be allowed to contain the separator character (`$`).

Reply:
clang already checks that the number of arguments is exactly 2 (see `test/SemaCXX/verbose-trap.cpp`).

I agree that we should check that `$` isn't in the strings.